server/app/files.go

```diff
@@ -86,7 +87,59 @@ func (a *App) GetFileInfo(filename string) (*mm_model.FileInfo, error) {
 	return fileInfo, nil
 }
 
+// ValidateFileOwnership checks if a file belongs to the specified board and team.
+func (a *App) ValidateFileOwnership(teamID, boardID, filename string) error {
+	fileInfo, err := a.GetFileInfo(filename)
+	if err != nil {
+		return err
+	}
+	if fileInfo != nil && fileInfo.Path != "" && fileInfo.Path != emptyString {
+		expectedPath := filepath.Join(teamID, boardID, filename)
+		if fileInfo.Path == expectedPath {
+			return nil
+		}
+		if err := a.validateFileReferencedByBoard(boardID, filename); err != nil {
```

Comment:
Can you explain why this check is needed if the above expected file patch check is `false`? `validateFileReferencedByBoard` fetches all blocks of a board and iterates through them which would involve fetching a large amount of data from database and processing it. Any way we can avoid it?

Reply:
Added the comment and created the ticket: https://mattermost.atlassian.net/browse/MM-64852